React/Recipes/Session Handling.md

```diff
@@ -100,15 +204,17 @@ const SomePrivatePage = () => {
 };
 
 const Content = () => {
-  const { state } = useAuth();
+  const { state } = useSession();
 
   const session = state.data;
 
   return {session ? <PrivateContent /> : <Loading />}
 }
 ```
 
-Although this might look like a simple page redirect, render will occur and might hurt app performance. Once the page is loaded and the component is mounted, `state.data` will be `undefined`. Once request to read a session, `state.data` or `state.error` will be defined, and re-render will occur. This means that all child components will be re-rendered, too, which can potentially be an expensive operation. To optimize this, we'll create a component that will handle the redirect.
+Although this might look like a simple page redirect, it's not really optimized - rerender will occur and might hurt performance.
```

Comment:
I'm not sure if it's a good idea to show an example that is not good enough, at least not as a first example.

Reply:
How can we demonstrate this pitfall in a better way? I often see this in our projects, and I don't think our devs knows about this pitfall.